### PR TITLE
Home Screen: Add Store Management (Quick Links) card

### LINKED ISSUE #4082: Quick Links (Manage your store) card
**User story:**
As a new merchant
I want to easily access specific areas when I finish the setup
So I can quickly fine-tune my store

quick-links

**Description:** we want to implement a simple way to redirect merchants to specific parts of WooCommerce after they complete or skip the store setup flow. This first version will be mainly focused on tasks for new merchants, but we think that some links can also be useful for existing users, so we'll roll it out to all the users.
In the future, we'll evaluate if we should tailor the quick links to different segments. This will depend on the adoption rate of this first version.

**Acceptance criteria:**
- [ ] When the user enters the home screen he will see the "Quick links" card
- [ ] The user can access each area by clicking on the chevron icon or in the title
- [ ] When the user clicks "Market my store" he will be redirected to WooCommerce - Marketing
- [ ] When the user clicks "Add products" he will be redirected to WooCommerce - Products
- [ ] When the user clicks "Personalize my store" he will be redirected to Appearance - Customize
- [ ] When the user clicks "Shipping settings" he will be redirected to WooCommerce - Settings - Shipping
- [ ] When the user clicks "Tax settings" he will be redirected to WooCommerce - Settings - Tax
- [ ] When the user clicks "Payment settings" he will be redirected to WooCommerce - Settings - Payments
- [ ] When the user clicks "Edit store details" he will be redirected to WooCommerce - Settings - General
- [ ] When the user clicks "View my store" he will be redirected to the site frontend

**Event tracking:**
Event name: wcadmin_home_quick_links_click
Event properties:
task_name (the name of each link that was clicked. Eg: Marketing, Add Products, etc)

**Additional links:**
P2 with final design: pbIJXs-4Z-p2
Designs https://www.figma.com/file/Z2PiNnkhjrUIiR7IrfJRnU/Home-Screen?node-id=970%3A0

### PR DESCRIPTION
**Pull request by mattsherman**
Fixes woocommerce/woocommerce-admin#4082

This PR adds the new "Store Management" (Quick Links) card to the home screen.

### Accessibility

- [x] I've tested using only a keyboard (no mouse)
- [ ] I've tested using a screen reader
- [x] All animations respect [`prefers-reduced-motion`](https://developer.mozilla.org/en-US/docs/Web/CSS/@media/prefers-reduced-motion)
- [x] All text has [at least a 4.5 color contrast with its background](https://webaim.org/resources/contrastchecker/)

### Screenshots

ScreenCapture at Sat May 16 06:44:50 EDT 2020

### Detailed test instructions

- Enable debugging of Tracks in your browser console:

```
localStorage.setItem( 'debug', 'wc-admin:*' );
```

- View the new home screen.
- Verify that the "Store Management" card is shown.
- Click items and verify that they navigate to the correct locations (from woocommerce/woocommerce-admin#4082) 
    - When the user clicks "Market my store" he will be redirected to WooCommerce - Marketing
    - When the user clicks "Add products" he will be redirected to WooCommerce - Products
    - When the user clicks "Personalize my store" he will be redirected to Appearance - Customize
    - When the user clicks "Shipping settings" he will be redirected to WooCommerce - Settings - Shipping
    - When the user clicks "Tax settings" he will be redirected to WooCommerce - Settings - Tax
    - When the user clicks "Payment settings" he will be redirected to WooCommerce - Settings - Payments
    - When the user clicks "Edit store details" he will be redirected to WooCommerce - Settings - General
    - When the user clicks "View my store" he will be redirected to the site frontend
- Verify that `wcadmin_home_quick_links_click` Tracks events are recorded, with the `task_name` custom prop set appropriately, every time an item in the list is clicked.

### Unit tests

```
npm test -- -i client/quick-links/test/index.js
```

### Known issues

- Icons do not match design. (/cc @jameskoster Where can I find the correct icons to use?)
- Card always shows on home screen. It should only appear when the "Set up your store and start selling" task list card has been completed or manually dismissed. This will be addressed after the task list card has been merged (#4378)
- Card has incorrect ARIA roles (#4387). This should be addressed separately, as it is a problem with the underlying `<List>` component.
- Support link always goes to premium support. This should be addressed separately, as it is a core problem with support links in general (https://github.com/woocommerce/woocommerce/issues/25442), and a solution should be implemented to make sure that support links are consistent throughout the product.
- Storybook does not show styles correctly for `<QuickLinks>` component

### Changelog Note:

Enhancement: New "Store Management" quick links card on WooCommerce home screen.

### CODE REVIEW

**Comment by psealock** [comment]
I went the items to test and just noting it down here while I continue the review.

 - [x] Verify that the "Store Management" card is shown.
 - [x] When the user clicks "Market my store" he will be redirected to WooCommerce - Marketing
 - [x] When the user clicks "Add products" he will be redirected to WooCommerce - Products
 - [x] When the user clicks "Personalize my store" he will be redirected to Appearance - Customize
 - [x] When the user clicks "Shipping settings" he will be redirected to WooCommerce - Settings - Shipping
 - [x] When the user clicks "Tax settings" he will be redirected to WooCommerce - Settings - Tax
 - [x] When the user clicks "Payment settings" he will be redirected to WooCommerce - Settings - Payments
 - [x] When the user clicks "Edit store details" he will be redirected to WooCommerce - Settings - General
 - [x] When the user clicks "View my store" he will be redirected to the site frontend
 - [x] Verify that wcadmin_home_quick_links_click Tracks events are recorded, with the task_name custom prop set appropriately, every time an item in the list is clicked.

File: client/homepage/style.scss [hunk 1]
```diff
@@ -13,18 +13,41 @@
...
-		@include breakpoint('<782px') {
+		@include breakpoint( '<782px' ) {
 			width: 100%;
 			position: inherit;
 			top: auto;
 		}
+
+		& > * {
+			margin-bottom: 24px;
+		}
```
Comment (psealock) [inline client/homepage/style.scss:24]: 👍

File: client/homepage/style.scss [hunk 2]
```diff
@@ -13,18 +13,41 @@
...
 
-	@include breakpoint('<782px') {
+	@include breakpoint( '<782px' ) {
 		margin-left: -16px;
 		margin-right: -16px;
 		flex-direction: column-reverse;
 	}
+
+	.components-card {
+		.components-card__header.is-size-large {
```
Comment (psealock) [inline client/homepage/style.scss:34]: Are these rules going to apply to all large cards on the home page or just this one? If just this one, maybe something specific to the quicklinks would be best.

File: client/homepage/style.scss [hunk 3]
```diff
@@ -13,18 +13,41 @@
...
+			display: grid;
+			//align-items: center;
+			min-height: 63px;
+			padding: 13px 16px;
```
Comment (psealock) [inline client/homepage/style.scss:38]: We should stick with `$gap` and `$gap-small` if possible.

File: client/homepage/style.scss [hunk 4]
```diff
@@ -13,18 +13,41 @@
...
+	@include breakpoint( '<782px' ) {
 		margin-left: -16px;
 		margin-right: -16px;
 		flex-direction: column-reverse;
 	}
+
+	.components-card {
+		.components-card__header.is-size-large {
+			display: grid;
+			//align-items: center;
```
Comment (psealock) [inline client/homepage/style.scss:36]: errant commented out code

File: client/quick-links/index.js [hunk 5]
```diff
@@ -0,0 +1,190 @@
...
+			title: __( 'View my store', 'woocommerce-admin' ),
+			type: 'external',
+			href: props.getSetting( 'siteUrl' ),
+			iconName: 'external',
+			listItemTag: 'view-store',
+		},
+	];
+}
+
+function onItemClick( props, event ) {
```
Comment (psealock) [inline client/quick-links/index.js:89]: How about `handleOnItemClick` since `onItemClick` is already a prop?

File: client/quick-links/index.js [hunk 6]
```diff
@@ -0,0 +1,190 @@
...
+
+	if ( iconName.startsWith( 'gridicons-' ) ) {
+		icon = <Gridicon icon={ iconNameWithoutPrefix } />;
+	} else if ( iconName.startsWith( 'dashicons-' ) ) {
+		icon = <Icon icon={ iconNameWithoutPrefix } />;
+	} else {
+		icon = <Icon icon={ iconName } />;
+	}
+
+	return icon;
```
Comment (psealock) [inline client/quick-links/index.js:125]: Looks like a really handy util function. I wonder if we could use this elsewhere?

File: client/quick-links/index.js [hunk 7]
```diff
@@ -0,0 +1,190 @@
...
+
+function getListItems( props ) {
+	return getItems( props ).map( ( item ) => {
+		return {
+			title: item.title,
+			before: getIcon( item.iconName ),
+			after: getIcon( 'arrow-right-alt2' ),
+			...getLinkTypeAndHref( item ),
+			listItemTag: item.listItemTag,
+			onClick: partial( onItemClick, props ),
```
Comment (psealock) [inline client/quick-links/index.js:165]: Nice, just noting here that we did some investigation already to determine if this function gets called before page navigation and in particular if Tracks events used in this way do indeed get recorded. It works as it should 😄

File: client/quick-links/index.js [hunk 8]
```diff
@@ -0,0 +1,190 @@
...
+	} );
+}
+
+const QuickLinks = ( props ) => {
+	const listItems = getListItems( props );
+
+	return (
+		<Card size="large" className="woocommerce-quick-links">
+			<CardHeader>
+				<h2>{ __( 'Store management', 'woocommerce-admin' ) }</h2>
```
Comment (psealock) [inline client/quick-links/index.js:176]: There is a handy container for heading levels to automatically frame out accessible heading hierarchy, `<H />`.

https://github.com/woocommerce/woocommerce-admin/blob/577d362f5149f86f68500a8fdd71ed4a9149065a/packages/components/src/section/index.js#L22-L31

File: client/quick-links/style.scss [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+	background-color: $background-color;
+
+	.components-card__body.is-size-large {
+		padding: 0;
+
+		.woocommerce-list {
+			margin: 0;
+			padding: 0;
+
+			.woocommerce-list__item {
```
Comment (psealock) [inline client/quick-links/style.scss:13]: These selectors are 4 and 5 levels deep, any way to make it less?

File: client/quick-links/style.scss [hunk 10]
```diff
@@ -0,0 +1,70 @@
...
+				// Ideally, we'd get these from `@wordpress/base-styles`.
+				// However, currently we are on an old version of `@wordpress/base-styles`,
+				// and updating that causes compatibility issues with the version of
+				// `@wordpress/components` that we are using, and updating that introduces
+				// breaking changes that we would need to address elsewhere in the
+				// `woocommerce-admin` codebase.
```
Comment (psealock) [inline client/quick-links/style.scss:19]: Great documentation! Should we add to it by creating an issue and linking it here?

**Review by psealock — COMMENTED** [review]
This is looking great @mattsherman and testing well ! Just some comments about a few relatively minor things.

**Review by jameskoster — DISMISSED** [review]
@mattsherman the icons are a part of the [WordPress core icons package](https://github.com/WordPress/gutenberg/tree/master/packages/icons) and should be accessible via the `Icon` component. :)

File: client/homepage/style.scss (hunk 3 above)
Comment (jameskoster) [reply client/homepage/style.scss:38]: Is it possible to use the [`Card`](https://wordpress.github.io/gutenberg/?path=/story/components-card--default) component here? Then we don't have to worry about things like padding values.

File: client/homepage/style.scss (hunk 2 above)
Comment (mattsherman) [reply client/homepage/style.scss:34]: I'd expect we'd want to apply it to all large cards on the home page, for consistency.

File: client/homepage/style.scss (hunk 3 above)
Comment (mattsherman) [reply client/homepage/style.scss:38]: I'm already using the `@wordpress/components` `Card` component. But, the resulting alignment without tweaking things, differs from both the `@woocommerce/components` `Card` component and the design Figma.

I've updated the code to use the gap variables, as suggested.

File: client/homepage/style.scss (hunk 4 above)
Comment (mattsherman) [reply client/homepage/style.scss:36]: Oops. Removed!

File: client/homepage/style.scss (hunk 2 above)
Comment (mattsherman) [reply client/homepage/style.scss:34]: Further, I'd expect that all cards on the home page should use the `@wordpress/components` `Card`.

File: client/quick-links/index.js (hunk 5 above)
Comment (mattsherman) [reply client/quick-links/index.js:89]: Will update.

File: client/quick-links/style.scss (hunk 10 above)
Comment (mattsherman) [reply client/quick-links/style.scss:19]: I actually created an issue last week. I'll add a link to it in the code comment!

File: client/quick-links/index.js (hunk 8 above)
Comment (mattsherman) [reply client/quick-links/index.js:176]: Will use, thanks for letting me know!

**Comment by mattsherman** [comment]
@jameskoster All of the icons used in the design (https://www.figma.com/file/Z2PiNnkhjrUIiR7IrfJRnU/Home-Screen?node-id=970%3A0) are not available in the `Icon` component. And, those that do differ in style from what you used.

It would be very helpful if you could provide me with the name of each icon to use, so that I can ensure that things are implemented as desired. Right now, I'm using a combination of `Icon` (and therefore `dashicon`) and `gridicons` icons to pick things that look close to what the design showed.

File: client/quick-links/index.js (hunk 6 above)
Comment (mattsherman) [reply client/quick-links/index.js:125]: Agreed. But, depending on what ends up happening in regards to what icons are used, we might no need this function for this component. If we end up keeping it, I can refactor it out to a common component/function.

File: client/quick-links/style.scss (hunk 9 above)
Comment (mattsherman) [reply client/quick-links/style.scss]: I've improved this a little bit.

**Comment by mattsherman** [comment]
@psealock Thanks for the review and feedback! I've addressed your feedback in the latest commits.

If you feel the PR is ready to be approved outside of the icon changes, I can always take care of those in a followup PR.

File: client/homepage/style.scss (hunk 2 above)
Comment (psealock) [reply client/homepage/style.scss:34]: Cool, I saw that issue 👍

**Review by psealock — APPROVED** [review]
This is looking fantastic and thanks for the changes. LGTM and good to follow up with icons separately.

**Comment by jameskoster** [comment]
> All of the icons used in the design are not available in the Icon component. And, those that do differ in style from what you used.

Hm, is that the core Icon component? They should be there because I [added](https://github.com/WordPress/gutenberg/pull/21209) [them](https://github.com/WordPress/gutenberg/pull/21825). 

At least, I added the ones that I thought were missing, maybe some others need updating from their Dashicon counterparts 🤔 

Here's the explicit list of icons to use;

- Market my store - [Megaphone](https://github.com/WordPress/gutenberg/blob/master/packages/icons/src/library/megaphone.js)
- Add products - [Box](https://github.com/WordPress/gutenberg/blob/master/packages/icons/src/library/box.js)
- Personalize my store - [Brush](https://github.com/WordPress/gutenberg/blob/master/packages/icons/src/library/brush.js)
- Shipping settings - [Shipping](https://github.com/WordPress/gutenberg/blob/master/packages/icons/src/library/shipping.js)
- Tax settings - [Percent](https://github.com/WordPress/gutenberg/blob/master/packages/icons/src/library/percent.js)
- Payment settings - [Payment](https://github.com/WordPress/gutenberg/blob/master/packages/icons/src/library/payment.js)
- Edit store details - [Pencil](https://github.com/WordPress/gutenberg/blob/master/packages/icons/src/library/pencil.js)
- Get support - [Lifesaver](https://github.com/WordPress/gutenberg/blob/master/packages/icons/src/library/lifesaver.js)
- View my store - [External](https://github.com/WordPress/gutenberg/blob/master/packages/icons/src/library/external.js)

It looks like the Brush and Pencil icons will need to be updated to match the visual design of the others, but we'll handle that in core :)

Sorry for the confusion here.

**Comment by mattsherman** [comment]
> > All of the icons used in the design are not available in the Icon component. And, those that do differ in style from what you used.
> 
> Hm, is that the core Icon component? They should be there because I [added](https://github.com/WordPress/gutenberg/pull/21209) [them](https://github.com/WordPress/gutenberg/pull/21825).
> 
> At least, I added the ones that I thought were missing, maybe some others need updating from their Dashicon counterparts 🤔
> 
> Here's the explicit list of icons to use;
> [...]
> Sorry for the confusion here.

@jameskoster Thanks for the links to the icons to use. I think the confusion stemmed from my use of `Icon` from `@wordpress/components` instead of from `@wordpress/icons`. I've created a new issue to address updating the icons (#4396), so that I can merge this PR. I'll get a PR up for that and make sure to request a review from you on it so we can ensure that the correct icons are used!

**Pull request merged**